### PR TITLE
executor: little improvement for multi-delete.

### PR DESCRIPTION
**Pull request by winoros**
Some calculation can be moved outside the for loop.
PTAL @hanfei1991 @coocood @lamxTyler

### CODE REVIEW

File: executor/write.go [hunk 1]
```diff
@@ -193,6 +200,18 @@ func (e *DeleteExec) deleteMultiTables() error {
 	for _, t := range e.Tables {
 		tblMap[t.TableInfo.ID] = append(tblMap[t.TableInfo.ID], t)
 	}
+	var colPosInfos []tblColPosInfo
+	for id, cols := range e.SelectExec.Schema().TblID2Handle {
```
Comment (shenli) [inline executor/write.go]: Could you add comment for the loop?

File: executor/write.go [hunk 2]
```diff
@@ -193,6 +200,18 @@ func (e *DeleteExec) deleteMultiTables() error {
...
+		tbl := e.tblID2Table[id]
+		for _, col := range cols {
+			if names, ok := tblMap[id]; !ok || !isMatchTableName(names, col) {
```
Comment (coocood) [inline executor/write.go]: `isMatchTableName` is only used in this function, so this name is too general.
I think make it a method of `DeleteExec` and move rename to `isDeletingTable` is better.

```
if !e.isDeletingTable(tbleMap, col) {
    continue
}
```
Would be more clear.

**Comment by winoros** [comment]
@shenli @coocood PTAL

**Comment by shenli** [comment]
LGTM

**Pull request merged**